Framework/src/Framework/Logger.cxx

```diff
@@ -116,6 +142,47 @@ void close() {
   return;
 }
 
+Formatter& Formatter::get() {
+  static Formatter the_formatter;
+  return the_formatter;
+}
+
+void Formatter::set(int n) { Formatter::get().event_number_ = n; }
+
+void Formatter::operator()(const log::record_view& view,
+                           log::formatting_ostream& os) {
+  os << "[ " << log::extract<std::string>("Channel", view) << " ] "
+     << event_number_ << " ";
+  /**
+   * We _copy_ the value out of the log into our own type
+   * so that we can compare and convert it into a string.
+   * I expect this copying to be okay since its just a
+   * enum (int equivalent), but its good to be clear
+   */
+  level msg_level = *log::extract<level>("Severity", view);
+  switch (msg_level) {
+    case level::debug:
+      os << "debug";
+      break;
+    case level::info:
+      os << "info ";
+      break;
+    case level::warn:
+      os << "warn ";
```

Comment:
Extremely minor suggestion, but speaking of parsing, it could be important
```suggestion
      os << "info";
      break;
    case level::warn:
      os << "warn";
```

Reply:
So I left the spaces in on purpose so that the message label is constant width. Do we want to keep that constant width? (And maybe apply a width to the event number and processor names) or not worry about constant width

Reply:
I see, well with the `debug` the printout is `debug:` while with the info that becomes `info :` which looked a bit weird. And indeed with the processor name being not the same length I didnt even think of the reason for having a space is the same length. I dont think it's realistic to have a fixed length for the processors, sometimes we have very long names, so I'd just give up on this.

Reply:
I guess it's true that it does make it nicer inside a given processor
```
[ ElectronCounter ] 9838 debug: Found 1 electrons (tracks) using input collection TriggerPadTracksY_
[ trigger ] 9838 debug: Got trigger energy cut 1500 for 1 electrons counted in the event.
[ trigger ] 9838 debug: Got trigger energy sum 966.672; and decision is pass = true
[ ecalVeto ] 9839 debug: Straight tracks found (before merge): 0
[ ecalVeto ] 9839 debug: Beginning track merging using 0 tracks
[ ecalVeto ] 9839 debug: Straight tracks found (after merge): 0
[ ecalVeto ] 9839 debug: Finding linreg tracks from 4 hits
[ ecalVeto ] 9839 info :   MIP tracking completed; found 0 straight tracks and 0 lin-reg tracks
[ ecalVeto ] 9839 debug:   The pred > bdtCutVal = true
[ ElectronCounter ] 9839 debug: Found 1 electrons (tracks) using input collection TriggerPadTracksY_
```

Reply:
So what if we put the space in front of the `info` and `warn`? That will satisfy both the fixed length, and will not introduce a space between the text and the colon

Reply:
sounds like an easy way out